src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -16,19 +34,24 @@ export const TodoFilter = () => (
         type="text"
         className="input"
         placeholder="Search..."
+        value={query}
+        onChange={(event) => ChangeQuery(event.target.value)}
       />
       <span className="icon is-left">
         <i className="fas fa-magnifying-glass" />
       </span>
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
-        />
-      </span>
+      {!!query.length && (
```

Comment:
Double !! ?
```suggestion
      {query.length && (
```
<img width="824" alt="image" src="https://user-images.githubusercontent.com/57747481/186610394-f3302564-2434-4bfb-8959-e851717a258a.png">